app/controllers/concerns/application_notifications_concern.rb

```diff
@@ -0,0 +1,14 @@
+module ApplicationNotificationsConcern
+  extend ActiveSupport::Concern
+
+  # Returns the view path of the notification
+  #
+  # @param [Course::Notification|UserNotification] notification The target notification
+  # @return [String] The view path of the notification
+  def notification_view_path(notification)
+    activity = notification.activity
+    root_path = "#{activity.notifier_type}/#{activity.event}"
+    notification_class_name = notification.class.name.underscore.sub('/', '_').pluralize
```

Comment:
I'm so sorry, I pushed this version last night just for checking how many comments I've fixed. And I know why to use `gsub`. So sorry I have not fixed last night... :flushed: